debug.go

```diff
@@ -0,0 +1,71 @@
+package anonymization
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"path"
+	"runtime"
+)
+
+var (
+	stderr *bufio.Writer = nil
+
+	// full path or only file name (function name)?
+	fullPath = false
+
+	// control debugging only from this file
+	dbgAllowed = true
+
+	// control debugging by calling turnDbgOn() on a per function basis
+	dbgFlag = false
+
+	callers [1]uintptr
+)
+
+func DbgOn() bool {
+	prev := dbgFlag
+	if dbgAllowed {
+		dbgFlag = true
+	}
+	return prev
+}
+
+func DbgRestore(dbg bool) {
+	dbgFlag = dbg
+}
+
+func DbgOff() {
+	dbgFlag = false
+}
+
+func Dbg(format string, args ...interface{}) {
+	if dbgFlag {
+		if stderr == nil {
+			stderr = bufio.NewWriter(os.Stderr)
+		}
+		msg := fmt.Sprintf(format, args...)
+		defer func() {
+			fmt.Fprintln(stderr, msg)
+			stderr.Flush()
+		}()
+		n := runtime.Callers(2, callers[:])
+		if n == 0 {
+			fmt.Fprintf(stderr, format, args...)
```

Comment:
wouldn't this print twice the message, since you have the defer Fprintln(msg) above?

Reply:
yes. I forgot to remove it after adding the `defer`.